contrib/jms/src/main/java/com/google/cloud/dataflow/contrib/jms/JmsIO.java

```diff
@@ -0,0 +1,532 @@
+                this.coder = coder;
+                this.maxNumMessages = maxNumMessages;
+            }
+
+            /**
+             * Returns a transform that's like this one but with the given step name.
+             *
+             * @param name the step name
+             * @return the Bound
+             */
+            public Bound<T> named(String name) {
+                return new Bound<T>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            /**
+             * Returns a transform that's like this one but using the JMS connection factory.
+             *
+             * @param connectionFactory the JMS connection factory to use
+             * @return the Bound
+             */
+            public Bound<T> connectionFactory(ConnectionFactory connectionFactory) {
+                return new Bound<T>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            /**
+             * Returns a transform that's like this one but that reads
+             * from the specified JMS queue.
+             *
+             * @param queue the JMS queue
+             * @return the Bound
+             */
+            public Bound<T> queue(String queue) {
+                return new Bound<T>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            /**
+             * Returns a transform that's like this one but that reads
+             * from the specified JMS topic.
+             *
+             * @param topic the JMQ topic
+             * @return the Bound
+             */
+            public Bound<T> topic(String topic) {
+                return new Bound<T>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            /**
+             * Returns a transform that's like this one but that uses the
+             * given {@link Coder} to decode each record into a value of
+             * type {@code X}.
+             *
+             * @param coder the coder to use
+             * @return the Bound
+             */
+            public <X> Bound<X> withCoder(Coder<X> coder) {
+                return new Bound<>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            /**
+             * Return a transform that's like this one but will only read up
+             * to the specified maximum number of messages from the JMS broker.
+             * The transform produces a <i>bounded</i> {@link PCollection}.
+             *
+             * @param maxNumMessages the max number messages limit to consume
+             * @return the Bound
+             */
+            public Bound<T> maxNumMessages(int maxNumMessages) {
+                return new Bound<T>(name, connectionFactory, queue, topic, coder, maxNumMessages);
+            }
+
+            @Override
+            public PCollection<T> apply(PInput input) {
+                if (connectionFactory == null) {
```

Comment:
you can put this code in [`PTransform#validate`](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/blob/master/sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/PTransform.java#L195) instead.

And since you link in guava, you can feel free to do

```java
import static ...Preconditions.checkNotNull;
...
checkNotNull(connectionFactory, "message here");
```

to simplify.